test/Datadog.Trace.Tests/TracerSettingsTests.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Sampling;
+using Moq;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.Tests
+{
+    public class TracerSettingsTests
+    {
+        private readonly Mock<IAgentWriter> _writerMock;
+        private readonly Mock<ISampler> _samplerMock;
+
+        public TracerSettingsTests()
+        {
+            _writerMock = new Mock<IAgentWriter>();
+            _samplerMock = new Mock<ISampler>();
+        }
+
+        [Theory]
+        [InlineData(ConfigurationKeys.Environment, Tags.Env, null)]
+        [InlineData(ConfigurationKeys.Environment, Tags.Env, "custom-env")]
+        [InlineData(ConfigurationKeys.ServiceVersion, Tags.Version, null)]
+        [InlineData(ConfigurationKeys.ServiceVersion, Tags.Version, "custom-version")]
+        public void ConfiguredTracerSettings_DefaultTagsSetFromEnvironmentVariable(string environmentVariableKey, string tagKey, string value)
+        {
+            // save original value so we can restore later
+            var originalValue = Environment.GetEnvironmentVariable(environmentVariableKey);
+
+            Environment.SetEnvironmentVariable(environmentVariableKey, value, EnvironmentVariableTarget.Process);
+            IConfigurationSource source = new EnvironmentConfigurationSource();
```

Comment:
Should we pass a custom `IConfigurationSource` here that doesn't rely on environment variables? Maybe `NameValueConfigurationSource`? It feels like this test should need to set environment variables, and reading setting from env vars should be tested elsewhere.

Reply:
Leaving this for a future PR.